src/oidcendpoint/authz/__init__.py

```diff
@@ -14,50 +17,43 @@ class AuthzHandling(object):
     def __init__(self, endpoint_context, **kwargs):
         self.endpoint_context = endpoint_context
         self.cookie_dealer = endpoint_context.cookie_dealer
-        self.permdb = {}
         self.kwargs = kwargs
+        self.grant_config = kwargs.get("grant_config", {})
 
-    def __call__(self, *args, **kwargs):
-        return ""
-
-    def set(self, uid, client_id, permission):
-        try:
-            self.permdb[uid][client_id] = permission
-        except KeyError:
-            self.permdb[uid] = {client_id: permission}
+    def __call__(self, user_id: str, client_id: str, request: Union[dict, Message],
+                 resources: Optional[list] = None) -> Grant:
+        args = self.grant_config.copy()
 
-    def permissions(self, cookie=None, **kwargs):
-        if cookie is None:
-            return None
-        else:
-            logger.debug("kwargs: %s" % sanitize(kwargs))
+        scope = request.get("scope")
+        if scope:
+            args["scope"] = scope
 
-            val = self.cookie_dealer.get_cookie_value(cookie)
-            if val is None:
-                return None
-            else:
-                b64, _ts, typ = val
+        claims = request.get("claims")
+        if claims:
+            args["claims"] = claims.to_dict()
 
-            info = cookie_value(b64)
-            return self.get(info["sub"], info["client_id"])
+        if resources is None:
+            args["resources"] = [client_id]
+        else:
+            args["resources"] = resources
 
-    def get(self, uid, client_id):
-        try:
-            return self.permdb[uid][client_id]
-        except KeyError:
-            return None
+        grant = Grant(**args)
+        # This is where user consent should be handled
+        for interface in ["userinfo", "introspection", "id_token", "token"]:
+            grant.claims[interface] = self.endpoint_context.claims_interface.get_claims(
+                client_id=client_id, user_id=user_id, scopes=request["scope"], usage=interface
+            )
+        return grant
 
 
 class Implicit(AuthzHandling):
-    def __init__(self, endpoint_context, permission="implicit"):
-        AuthzHandling.__init__(self, endpoint_context)
-        self.permission = permission
-
-    def permissions(self, cookie=None, **kwargs):
-        return self.permission
+    def __init__(self, endpoint_context, **kwargs):
+        AuthzHandling.__init__(self, endpoint_context, **kwargs)
```

Comment:
Not needed

Reply:
Matter of programming style. But yes, not needed.